src/Api/Endpoints/Candidate/Visit/Image/Format/Raw.php

```diff
@@ -0,0 +1,162 @@
+ * @link     https://github.com/aces/Loris
+ */
+namespace LORIS\Api\Endpoints\Candidate\Visit\Image\Format;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Api\Endpoint;
+
+/**
+ * This class handles request to an image raw format.
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+class Raw extends Endpoint implements \LORIS\Middleware\ETagCalculator
+{
+
+    /**
+     * The requested Image
+     *
+     * @var \LORIS\Image
+     */
+    private $_image;
+
+    /**
+     * Contructor
+     *
+     * @param \LORIS\Image $image The requested image
+     */
+    public function __construct(\LORIS\Image $image)
+    {
+        $this->_image = $image;
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return array('GET');
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return array(
+                'v0.0.2',
+                'v0.0.3-dev',
+               );
+    }
+
+    /**
+     * Delegates the request to a private handler appropriate to the requets's
+     * method.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+
+        case 'OPTIONS':
+            return (new \LORIS\Http\Response())
+                ->withHeader('Allow', $this->allowedMethods());
+
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+
+        }
+    }
+
+    /**
+     * Create an array representation of this endpoint's reponse body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        $info = $this->_image->getFileInfo();
+
+        if (!$info->isFile()) {
+            error_log('file in database but not in file system');
+            return new \LORIS\Http\Response\NotFound();
+        }
+
+        if (!$info->isReadable()) {
+            error_log('file exist but is not readable by webserver');
+            return new \LORIS\Http\Response\NotFound();
+        }
+
+        if (!`which minctoraw`) {
```

Comment:
I always get 'minctoraw not installed' here. Does it work for you?

I may be wrong but I doubt apache is sourcing the MINC tool kit `.sh` file therefore, `minctoraw` is never seen by apache and `which minctoraw` will never appear to be installed and therefore always return 'minctoraw not installed'. Am I missing something?

FYI, the MINC tools path is stored in `config.xml` under the `<MINCToolsPath>` tag. Path can be for example: `/opt/minc/`. The full path to minctoraw would then be `/opt/minc/bin/minctoraw` (a.k.a. `<MINCToolsPath>/bin/minctoraw`).

Reply:
I have the following lines in /etc/apache2/envvars
```
# Add minctoolkit to apache user environement 
export PATH=$PATH:/opt/minc/1.9.16/bin
```

Reply:
The line `which minctoraw` is from the old API code. I thought it was working at some point so I added the lines in envvars.

Reply:
mmm. I added the line in `envvars`. Is it documented somewhere that I missed at the time of install that this need to be added to the `envvars` file of apache? Wouldn't it be better to use the config setting in the `config.xml` file? 

Looks like BrainBrowser is using the config <MINCToolsPath> so maybe it is best to do the same for the API? What do you think?

Also @driusan, @johnsaigle , is there a reason why this setting MINCToolsPath is in the `config.xml` file and not in the frontend? Maybe a security reason? Just wondering.

Reply:
@cmadjar I will change it to use de config.xml
And I second your proposition of moving that in the database.

Reply:
Yes it should be in the DB. Not sure of the origin of why it's in that file. https://github.com/aces/Loris/pull/4023

Reply:
Maybe that path was forgotten when all the paths were moved to the Config module. I created an issue (#5019) for that. Should not be complicated to move it there. ;)